src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -841,6 +841,62 @@ public void It_marks_package_references_as_externally_resolved(bool? markAsExter
+        [InlineData("net6.0", false, true, true, "False,True")]     // predefinedCulturesOnly explicitly defined as true.
+        [InlineData("net6.0", true, false, false, "True,True")]     // predefinedCulturesOnly default value is true when Invariant is true
+        [InlineData("net6.0", true, false, true, "True,False")]     // predefinedCulturesOnly explicitly defined as false.
+        [InlineData("net6.0", true, true, true, "True,True")]       // predefinedCulturesOnly explicitly defined as true.
+        public void It_can_implicitly_define_predefined_Cultures_only(string targetFramework, bool invariantValue, bool predefinedCulturesOnlyValue, bool definePredefinedCulturesOnly, string expectedPredefinedCulturesOnlyValue)
+        {
+            var testProj = new TestProject()
+            {
+                Name = "CheckPredefineCulturesOnly",
+                TargetFrameworks = targetFramework,
+                IsExe = true,
+            };
+
+            testProj.AdditionalProperties["InvariantGlobalization"] = invariantValue ? "true" : "false";
+
+            if (definePredefinedCulturesOnly)
+            {
+                testProj.AdditionalProperties["PredefinedCulturesOnly"] = predefinedCulturesOnlyValue ? "true" : "false";
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProj, targetFramework);
+            File.WriteAllText(Path.Combine(testAsset.Path, testProj.Name, $"{testProj.Name}.cs"), @"
+                using System;
+                using System.Reflection;
+                class Program
+                {
+                    static void Main(string[] args)
+                    {
+                        bool invariant = false;
+                        bool predefinedCulturesOnly = false;
+                        try { invariant = (bool) typeof(object).Assembly.GetType(""System.Globalization.GlobalizationMode"").GetProperty(""Invariant"", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null); } catch {}
+                        try { predefinedCulturesOnly = (bool) typeof(object).Assembly.GetType(""System.Globalization.GlobalizationMode"").GetProperty(""PredefinedCulturesOnly"", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null); } catch {}
+
+                        Console.WriteLine($""{invariant},{predefinedCulturesOnly}"");
+                    }
+                }
+            ");
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.Path, testProj.Name));
+            buildCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var runCommand = new RunExeCommand(Log, Path.Combine(buildCommand.GetOutputDirectory(targetFramework).FullName, $"{testProj.Name}.exe"));
+            var stdOut = runCommand.Execute().StdOut.Split(Environment.NewLine.ToCharArray()).Where(line => !string.IsNullOrWhiteSpace(line));
+            stdOut.Should().BeEquivalentTo(expectedPredefinedCulturesOnlyValue);
```

Comment:
`BeEquivalentTo` ignores the ordering of the elements in the collections, which is probably not what you want.

Reply:
Would it matter if I have just one element in the collection? what you suggest to use instead?

Reply:
Never mind, I thought you were splitting on commas instead of newlines.  So the code you have works.